### PR TITLE
Pruning, also, cleaning up NameType traits a bit

### PR DESCRIPTION
**Pull request by dirvine**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/564)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @inetic

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/id.rs [hunk 1]
```diff
@@ -22,7 +22,6 @@ use NameType;
 // Note: name field is initially same as original_name, this should be later overwritten by
 // relocated name provided by the network using assign_relocated_name method
 // TODO (ben 2015-04-01) : implement order based on name
-#[derive(Clone)]
```
Comment (benjaminbollen) [inline src/id.rs]: I was planning on cloning the Id from the initial Routing object into the Routing core, as vaults acting on Routing need to sign messages as the RoutingNode; alternatively we need to have a separate Action:: that takes the elements of a message, but not a fully SignedMessage, and then finish construction in RoutingNode

**Comment by benjaminbollen** [comment]
looks good to me
raises few discussion points for Monday

**Comment by inetic** [comment]
@dirvine CI check are failing, is that expected?

**Pull request merged**